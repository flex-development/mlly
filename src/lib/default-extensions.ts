/**
 * @file defaultExtensions
 * @module mlly/lib/defaultExtensions
 */

import type { Ext } from '@flex-development/pathe'

/**
 * Default resolvable file extensions.
 *
 * @see {@linkcode Ext}
 *
 * @const {Set<Ext>} defaultExtensions
 */
const defaultExtensions: Set<Ext> = new Set([
  '.mjs',
  '.mts',
  '.cjs',
  '.cts',
  '.js',
  '.ts',
  '.jsx',
  '.tsx',
  '.css',
  '.json',
  '.node',
  '.wasm',
  '.d.mts',
  '.d.cts',
  '.d.ts'
])

export default defaultExtensions
