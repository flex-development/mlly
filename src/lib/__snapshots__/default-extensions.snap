// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:lib/defaultExtensions > should contain default extensions 1`] = `
Set {
  ".mjs",
  ".mts",
  ".cjs",
  ".cts",
  ".js",
  ".ts",
  ".jsx",
  ".tsx",
  ".css",
  ".json",
  ".node",
  ".wasm",
  ".d.mts",
  ".d.cts",
  ".d.ts",
}
`;
